File: lib/models/subscription.dart
enum SubscriptionType {
  monthly('monthly', 'Mensuel'),
  yearly('yearly', 'Annuel'),
  lifetime('lifetime', 'À vie');

  final String value;
  final String label;
  const SubscriptionType(this.value, this.label);
}

enum DonationStatus {
  pending('pending', 'En attente'),
  completed('completed', 'Terminée'),
  failed('failed', 'Échouée');

  final String value;
  final String label;
  const DonationStatus(this.value, this.label);
}

enum ReportStatus {
  pending('pending', 'En attente'),
  resolved('resolved', 'Résolu'),
  dismissed('dismissed', 'Rejeté');

  final String value;
  final String label;
  const ReportStatus(this.value, this.label);
}

enum SuggestionStatus {
  pending('pending', 'En attente'),
  approved('approved', 'Approuvée'),
  rejected('rejected', 'Rejetée');

  final String value;
  final String label;
  const SuggestionStatus(this.value, this.label);
}

class Subscription {
  final String id;
  final String userId;
  final bool isActive;
  final DateTime startDate;
  final DateTime? endDate;
  final SubscriptionType subscriptionType;

  Subscription({
    required this.id,
    required this.userId,
    required this.isActive,
    required this.startDate,
    this.endDate,
    required this.subscriptionType,
  });

  bool get isExpired {
    if (endDate == null) return false; // Lifetime subscription
    return DateTime.now().isAfter(endDate!);
  }
}

class Donation {
  final String id;
  final String userId;
  final double amount;
  final DateTime date;
  final DonationStatus status;

  Donation({
    required this.id,
    required this.userId,
    required this.amount,
    required this.date,
    required this.status,
  });
}

class BugReport {
  final String id;
  final String userId;
  final String title;
  final String description;
  final DateTime date;
  final ReportStatus status;

  BugReport({
    required this.id,
    required this.userId,
    required this.title,
    required this.description,
    required this.date,
    required this.status,
  });
}

class AcronymSuggestion {
  final String id;
  final String userId;
  final String acronym;
  final String? description;
  final DateTime date;
  final SuggestionStatus status;

  AcronymSuggestion({
    required this.id,
    required this.userId,
    required this.acronym,
    this.description,
    required this.date,
    required this.status,
  });
}

File: Le_DARIEL/Models/Subscription.swift
//
//  Subscription.swift
//  Le_DARIEL
//
//  Created by FAEI971 on 23/07/2025.
//

import Foundation

struct Subscription: Codable {
    let id: UUID
    let userId: UUID
    var isActive: Bool
    var startDate: Date
    var endDate: Date?
    var subscriptionType: SubscriptionType
    
    enum SubscriptionType: String, Codable {
        case monthly = "monthly"
        case yearly = "yearly"
        case lifetime = "lifetime"
    }
    
    var isExpired: Bool {
        guard let endDate = endDate else {
            return false // Lifetime subscription
        }
        return Date() > endDate
    }
}

struct Donation: Codable {
    let id: UUID
    let userId: UUID
    let amount: Double
    let date: Date
    var status: DonationStatus
    
    enum DonationStatus: String, Codable {
        case pending = "pending"
        case completed = "completed"
        case failed = "failed"
    }
}

struct BugReport: Codable, Identifiable {
    let id: UUID
    let userId: UUID
    let title: String
    let description: String
    let date: Date
    var status: ReportStatus
    
    enum ReportStatus: String, Codable {
        case pending = "pending"
        case resolved = "resolved"
        case dismissed = "dismissed"
    }
}

struct AcronymSuggestion: Codable, Identifiable {
    let id: UUID
    let userId: UUID
    let acronym: String
    let description: String?
    let date: Date
    var status: SuggestionStatus
    
    enum SuggestionStatus: String, Codable {
        case pending = "pending"
        case approved = "approved"
        case rejected = "rejected"
    }
}





